src/compiler/checker.ts

```diff
@@ -9134,6 +9134,26 @@ namespace ts {
                 if (type) {
                     return addOptionality(type, /*isProperty*/ false, isOptional);
                 }
+                if (isMethodDeclaration(func) && isClassLike(func.parent)
+                    && (noImplicitAny || isInJSFile(func.parent))) { // TODO: Might want to fiddle with restrictions here

```

Comment:
out of curiosity - what's the reasoning to only apply this logic with `noImplicitAny` (and in JS files)? 🤔

Reply:
1. noImplicitAny: untyped parameters will get an error at the same time they get `any`, so this change can't cause code with 0 errors to compile with >0 errors.
2. JS files: in JS, errors aren't reported anyway, so any improvement in typing should show up as improvement in completions. (this might need to change to exclude checkjs)